src/Microsoft.TestPlatform.CoreUtilities/Utilities/JobQueue.cs

```diff
@@ -140,7 +140,10 @@ public JobQueue(Action<T> processJob, string displayName, int maxQueueLength, in
             this.exceptionLogger = exceptionLogger;
 
             // Setup the background thread to process the jobs.
-            this.backgroundJobProcessor = Task.Run(() => this.BackgroundJobProcessor());
+            backgroundJobProcessor = new Thread(BackgroundJobProcessor);
```

Comment:
Why are we using Threads, we could also use Task with TaskCreationOptions.LongRunning, it is undocumented but under this Task gets its own thread. We can check this with CLR team.